### ISSUE TITLE
TESTS: Make sure to write to tempdir() everywhere

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/R.utils#107]
CRAN checks just reported:
```
Version: 2.9.1
Check: tests
Result: ERROR
     Running ‘Arguments-FILES.R’ [0s/1s]
     Running ‘FileProgressBar.R’ [0s/0s]
    Running the tests in ‘tests/FileProgressBar.R’ failed.
    Last 13 lines of output:
    
     The following objects are masked from 'package:base':
    
     cat, commandArgs, getOption, inherits, isOpen, parse, warnings
    
     >
     > # Creates a progress bar (of length 100) that displays it self as a file.
     > pb <- FileProgressBar("~/progress.simulation")
     Error in file(file, ifelse(append, "a", "w")) :
     cannot open the connection
     Calls: FileProgressBar ... update.FileProgressBar -> cat -> cat.default -> <Anonymous> -> file
     In addition: Warning message:
     In file(file, ifelse(append, "a", "w")) :
     cannot open file '/Users/urbanek/progress.simulation': Read-only file system
     Execution halted
Flavor: r-oldrel-osx-x86_64
```

**Comment by HenrikBengtsson** [comment]
One can use [Singularity](https://sylabs.io/docs/)  toreproduce this locally in a Linux container.  In my case I run this on Ubuntu 18.04.

## 1. Build R container
```r
$ singularity build r-base.simg docker://rocker/r-base
...
$ singularity exec r-base.simg Rscript --version
R scripting front-end version 3.6.1 (2019-07-05)
```

## 2. Build R package (using host R)

```r
$ Rscript --version
R scripting front-end version 3.6.1 (2019-07-05)

$ R CMD build R.utils
* checking for file 'R.utils/DESCRIPTION' ... OK
* preparing 'R.utils':
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building 'R.utils_2.9.1.tar.gz'
```

## 3. Check package (using container R and read-only /home)

Checking this package with `R CMD check --as-cran R.utils_2.9.1.tar.gz` on the host system gives all OK.  By running it in the above container with `/home` mounted as _read only_, we can reproduce the CRAN error:
```r
$ singularity exec -B /home:/home:ro r-base.simg R CMD check -o /tmp --as-cran R.utils_2.9.1.tar.gz
* using log directory '/tmp/R.utils.Rcheck'
* using R version 3.6.1 (2019-07-05)
* using platform: x86_64-pc-linux-gnu (64-bit)
* using session charset: UTF-8
* using option '--as-cran'
* checking for file 'R.utils/DESCRIPTION' ... OK
* this is package 'R.utils' version '2.9.1'
* checking CRAN incoming feasibility ...NB: need Internet access to use CRAN incoming checks
 Note_to_CRAN_maintainers
Maintainer: 'Henrik Bengtsson <henrikb@braju.com>'
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for .dll and .exe files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking serialization versions ... OK
* checking whether package 'R.utils' can be installed ... OK
* checking installed package size ... OK
* checking package directory ... OK
* checking for future file timestamps ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking use of S3 registration ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd line widths ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking examples ... OK
* checking for unstated dependencies in 'tests' ... OK
* checking tests ...
  Running ‘absolute-relative-paths.R’
  Running ‘Arguments-FILES.R’
  Running ‘attachLocally.R’
  Running ‘callHooks.R’
  Running ‘capitalize.R’
  Running ‘captureOutput.R’
  Running ‘cmdArgs.R’
  Running ‘colClasses.R’
  Running ‘commandArgs.R’
  Running ‘compressFile.R’
  Running ‘compressPDF.R’
  Running ‘copyRenameFile.R’
  Running ‘countLines.R’
  Running ‘cout.R’
  Running ‘createFileAtomically.R’
  Running ‘createLink.R’
  Running ‘dataFrame.R’
  Running ‘dimNA.R’
  Running ‘displayCode.R’
  Running ‘doCall.R’
  Running ‘eget.R’
  Running ‘egsub.R’
  Running ‘env.R’
  Running ‘extract.array.R’
  Running ‘fileAccess.R’
  Running ‘filePath.R’
  Running ‘FileProgressBar.R’
  Running ‘findFiles.R’
  Running ‘findSourceTraceback.R’
  Running ‘gcat.R’
  Running ‘gcDLLs.R’
  Running ‘getOption.R’
  Running ‘getParent.R’
  Running ‘GString.R’
  Running ‘hpaste.R’
  Running ‘insert.R’
  Running ‘intToHex.R’
  Running ‘isPackageLoaded.R’
  Running ‘isReplicated.R’
  Running ‘isUrl.R’
  Running ‘isZero.R’
  Running ‘Java.R’
  Running ‘listDirectory.R’
  Running ‘loadObject.R’
  Running ‘loadToEnv.R’
  Running ‘mkdirs.R’
  Running ‘mout.R’
  Running ‘mpager.R’
  Running ‘MultiVerbose.R’
  Running ‘nullfile.R’
  Running ‘NullVerbose.R’
  Running ‘Options.R’
  Running ‘parseRepos.R’
  Running ‘ProgressBar.R’
  Running ‘pushBackupFile.R’
  Running ‘pushTemporaryFile.R’
  Running ‘queryRCmdCheck.R’
  Running ‘readBinFragments.R’
  Running ‘readWindowsShellLink.R’
  Running ‘readWindowsShortcut.R’
  Running ‘resample.R’
  Running ‘seqToHumanReadable.R’
  Running ‘seqToIntervals.R’
  Running ‘Settings.R’
  Running ‘sourceDirectory.R’
  Running ‘sourceTo.R’
  Running ‘splitByPattern.R’
  Running ‘subplots.R’
  Running ‘symlinks,dirs.R’
  Running ‘symlinks,files.R’
  Running ‘System.R’
  Running ‘systemR.R’
  Running ‘tempvar.R’
  Running ‘TextStatusBar.R’
  Running ‘tmpfile.R’
  Running ‘toCamelCase.R’
  Running ‘touchFile.R’
  Running ‘use.R’
  Running ‘useRepos.R’
  Running ‘VComments.R’
  Running ‘Verbose.R’
  Running ‘whichVector.R’
  Running ‘withCapture.R’
  Running ‘withLocale.R’
  Running ‘withOptions.R’
  Running ‘withRepos.R’
  Running ‘withSeed.R’
  Running ‘withSink.R’
  Running ‘withTimeout.R’
  Running ‘wrap.array.R’
  Running ‘writeDataFrame.R’
  Running ‘zzz_finalizer_crash.R’
 ERROR
Running the tests in 'tests/FileProgressBar.R' failed.
Last 13 lines of output:
  The following objects are masked from 'package:base':
  
      cat, commandArgs, getOption, inherits, isOpen, nullfile, parse,
      warnings
  
  > 
  > # Creates a progress bar (of length 100) that displays it self as a file.
  > pb <- FileProgressBar("~/progress.simulation")
  Error in file(file, ifelse(append, "a", "w")) : 
    cannot open the connection
  Calls: FileProgressBar ... update.FileProgressBar -> cat -> cat.default -> <Anonymous> -> file
  In addition: Warning message:
  In file(file, ifelse(append, "a", "w")) :
    cannot open file '/home/hb/progress.simulation': Read-only file system
  Execution halted
* checking PDF version of manual ... WARNING
LaTeX errors when creating PDF version.
This typically indicates Rd problems.
* checking PDF version of manual without hyperrefs or index ... ERROR
Re-running with no redirection of stdout/stderr.
Hmm ... looks like a package
Error in texi2dvi(file = file, pdf = TRUE, clean = clean, quiet = quiet,  : 
  pdflatex is not available
Error in texi2dvi(file = file, pdf = TRUE, clean = clean, quiet = quiet,  : 
  pdflatex is not available
Error in running tools::texi2pdf()
You may want to clean up by 'rm -Rf /tmp/hb/Rtmp31XR3v/Rd2pdf5f30617166df'
* DONE

Status: 2 ERRORs, 1 WARNING
See '/tmp/R.utils.Rcheck/00check.log' for details.
```

**Comment by HenrikBengtsson** [comment]
This was fixed in commit 78c4b7b and
```r
$ singularity exec -B /home:/home:ro r-base.simg R CMD check -o /tmp --as-cran R.utils_2.9.1-9000.tar.gz
```
gives all OK.
